includes/class-wcpdf-main.php

```diff
@@ -941,10 +923,34 @@ public function set_phpmailer_validator( $mailArray ) {
 	/**
 	 * Logs the document creation to the order notes
 	 */
-	public function log_document_creation_to_order_notes( $document, $action ) {
-		if( ! empty( $document ) && ! empty( $action ) && ! empty( $order = $document->order ) ) {
-			$note = sprintf( __( 'PDF %s %s created on %s action.', 'woocommerce-pdf-invoices-packing-slips' ), $document->get_title() , $document->get_number()->get_plain(), $action );
-			$order->add_order_note( $note );
+	public function log_document_creation_to_order_notes( $document, $context ) {
+		if( ! empty( $document ) && ! empty( $context ) && isset( WPO_WCPDF()->settings->debug_settings['log_to_order_notes'] ) ) {
+			$message = __( 'PDF %s created on %s action.', 'woocommerce-pdf-invoices-packing-slips' );
+
+			// bulk document
+			if( $document->is_bulk ) {
+				foreach( $document->order_ids as $order_id ) {
+					$order          = wc_get_order( $order_id );
+					if( empty( $order ) ) continue;
+
+					$document_title = ucwords( str_replace( '-', ' ', $document->type ) );
+					$note           = sprintf( $message, $document_title, 'bulk' );
```

Comment:
here you're printing to a human string, so I would apply gettext to `bulk` (and maybe make it more descriptive, e.g.
`PDF %s created via %s` + `$document->get_title()` + `__('bulk action')`